### DISCUSSION TITLE
Storage folders

### DISCUSSION THREAD

**Discussion by maeddin** [appwrite/appwrite#2892] [Ideas]
Hi dear AppWrite community,
with the new update you can give permissions on bucket level. In addition to that, arbitrary folder structures within the buckets could be allowed. On the one hand, this would make it easier to keep an overview over all files, and on the other hand, it would facilitate operations such as deleting certain file groups. Furthermore, you could also allow permissions on folder level, which could be helpful in many use cases. This way you could create a folder per chat for a simple chat application and if you want to change the permissions of the files in this chat, you would not have to edit all files individually.
You could handle the permissions like this: If you want to access the file first/second/file.png. First it is looked whether one has these rights in 'first', then in 'second' and then with file.png. In this way you could also build the permission system hierarchically without changing the underlying system to the outside.

What is your opinion about this?

**Comment by n0obscertified** [comment]
this sounds awesome

**Comment by dch09** [comment]
bump. would love that feature
